dvc/repo/experiments/__init__.py

```diff
@@ -194,14 +194,47 @@ def _stash_exp(
                         if branch:
                             branch_name = ExpRefInfo.from_ref(branch).name
                         else:
-                            branch_name = ""
+                            branch_name = f"{resume_rev[:7]}"
                         if self.scm.is_dirty():
                             logger.info(
                                 "Modified checkpoint experiment based on "
                                 "'%s' will be created",
                                 branch_name,
                             )
                             branch = None
+                        elif (
+                            not branch
+                            or self.scm.get_ref(branch) != resume_rev
+                        ):
+                            msg = [
+                                (
+                                    "Nothing to do for unchanged checkpoint "
+                                    f"'{resume_rev[:7]}'. "
+                                )
+                            ]
+                            if branch:
+                                msg.append(
+                                    "To resume from the head of this "
+                                    "experiment, use "
```

Comment:
Is it obvious what "head" means? Sounds like a Git term. I'd be more explicit:

`"To resume this experiment as-is, apply the latest checkpoint: "`

Reply:
But there are more options right? So maybe

```
"Make changes to branch off from here (1234567) into a new experiment. "
"To continue this experiment as-is, apply the latest checkpoint with: ..."
```